tests/python/pants_test/backend/codegen/tasks/test_protobuf_integration.py

```diff
@@ -15,6 +15,12 @@
 
 class ProtobufIntegrationTest(PantsRunIntegrationTest):
 
+  def test_import_from_buildroot(self):
+    pants_run = self.run_pants(
+      ['gen.protoc', '--import-from-root',
+       'testprojects/src/protobuf/org/pantsbuild/testproject/import_from_buildroot/bar'])
+    self.assert_success(pants_run)
+
```

Comment:
Agreed, the isolation should come from the engine. Does the new engine admit anything like it easily?